Sources/Nimble/Matchers/BeCloseTo.swift

```diff
@@ -61,6 +62,7 @@ public class NMBObjCBeCloseToMatcher: NSObject, NMBMatcher {
         }
     }
 
+    @available(*, deprecated)
     @objc public func doesNotMatch(_ actualExpression: @escaping () -> NSObject?, failureMessage: FailureMessage, location: SourceLocation) -> Bool {
```

Comment:
Those are used for ObjC so should not be deprecated.

Reply:
Even so, if it's not deprecated, it shouldn't use deprecated types - is there any way we could avoid using `Matcher` inside?

Reply:
As stated, we should replace `matches` and `doesNotMatch` usages with `satisfies(_:_)`.

- https://github.com/Quick/Nimble/blob/70bc3fa8d9c2c054e20b9e1cd5ee6c72720c79e6/Sources/Nimble/Matchers/Predicate.swift#L203-L204
- https://github.com/Quick/Nimble/blob/70bc3fa8d9c2c054e20b9e1cd5ee6c72720c79e6/Sources/Nimble/Matchers/Predicate.swift#L210-L211

Reply:
Submitted #664 to address these.